op-node/node/node.go

```diff
@@ -221,21 +221,32 @@ func (n *OpNode) initRuntimeConfig(ctx context.Context, cfg *Config) error {
 			return eth.L1BlockRef{}, err
 		}
 
-		// Apply confirmation-distance
-		blNum := l1Head.Number
-		if blNum >= confDepth {
-			blNum -= confDepth
-		}
-		fetchCtx, fetchCancel = context.WithTimeout(ctx, time.Second*10)
-		confirmed, err := n.l1Source.L1BlockRefByNumber(fetchCtx, blNum)
-		fetchCancel()
-		if err != nil {
-			n.log.Error("failed to fetch confirmed L1 block for runtime config loading", "err", err, "number", blNum)
-			return eth.L1BlockRef{}, err
+		var safeBlock eth.L1BlockRef
+		if cfg.Driver.SequencerUseFinalized {
+			fetchCtx, fetchCancel = context.WithTimeout(ctx, time.Second*10)
+			safeBlock, err = n.l1Source.L1BlockRefByLabel(fetchCtx, eth.Finalized)
+			fetchCancel()
+			if err != nil {
+				n.log.Error("failed to fetch finalized L1 block for runtime config loading", "err", err)
+				return eth.L1BlockRef{}, err
+			}
+		} else {
+			// Apply confirmation-distance
+			blNum := l1Head.Number
+			if blNum >= confDepth {
+				blNum -= confDepth
+			}
+			fetchCtx, fetchCancel = context.WithTimeout(ctx, time.Second*10)
+			safeBlock, err = n.l1Source.L1BlockRefByNumber(fetchCtx, blNum)
+			fetchCancel()
+			if err != nil {
+				n.log.Error("failed to fetch confirmed L1 block for runtime config loading", "err", err, "number", blNum)
+				return eth.L1BlockRef{}, err
+			}
```

Comment:
Don't really know if it's going to make things clearer or even if it going to make the merges harder, but the cancel logic could be common for both
```suggestion
		var safeBlock eth.L1BlockRef
		fetchCtx, fetchCancel = context.WithTimeout(ctx, time.Second*10)
		if cfg.Driver.SequencerUseFinalized {
			safeBlock, err = n.l1Source.L1BlockRefByLabel(fetchCtx, eth.Finalized)
		} else {
			// Apply confirmation-distance
			blNum := l1Head.Number
			if blNum >= confDepth {
				blNum -= confDepth
			}
			safeBlock, err = n.l1Source.L1BlockRefByNumber(fetchCtx, blNum)
		}
		fetchCancel()
		if err != nil {
			if cfg.Driver.SequencerUseFinalized {
				n.log.Error("failed to fetch finalized L1 block for runtime config loading", "err", err)
			} else {
				n.log.Error("failed to fetch confirmed L1 block for runtime config loading", "err", err, "number", blNum)
			}
			return eth.L1BlockRef{}, err
		}
```
Also if we change the log to something understandable by both like would be cleaner. But it's only a nit, and we are not gaining too much either, so it's up to you